ggml/include/ggml-backend.h

```diff
@@ -90,51 +98,84 @@ extern "C" {
     GGML_API void ggml_backend_tensor_copy_async(ggml_backend_t backend_src, ggml_backend_t backend_dst, struct ggml_tensor * src, struct ggml_tensor * dst);
 
     // events
-    GGML_API ggml_backend_event_t   ggml_backend_event_new        (ggml_backend_t backend);
-    GGML_API void                   ggml_backend_event_free       (ggml_backend_event_t event);
-    GGML_API void                   ggml_backend_event_record     (ggml_backend_event_t event);
-    GGML_API void                   ggml_backend_event_synchronize(ggml_backend_event_t event);
-    GGML_API void                   ggml_backend_event_wait       (ggml_backend_t backend, ggml_backend_event_t event);
+    GGML_API ggml_backend_event_t ggml_backend_event_new        (ggml_backend_dev_t device);
+    GGML_API void                 ggml_backend_event_free       (ggml_backend_event_t event);
+    GGML_API void                 ggml_backend_event_record     (ggml_backend_event_t event, ggml_backend_t backend);
```

Comment:
`ggml_backend_t` represents a stream or async queue. The events are associated with a device, but not a stream. `ggml_backend_event_record` records the event on the stream represented by `backend`, which should be a backend of the same device than the event. I know that this is a bit confusing at the moment, `ggml_backend_t` should be renamed to something like `ggml_backend_stream`, but I am afraid that it will break a lot of code.